corona-warn-app/cwa-website
# Update blog test page (resolves #1991)

This PR resolves issue https://github.com/corona-warn-app/cwa-website/issues/1991.

In [cypress/fixtures/blog/2099-01-01-visual-testing](https://github.com/corona-warn-app/cwa-website/tree/master/cypress/fixtures/blog/2099-01-01-visual-testing) `index.md` and `index_de.md` the PR:

1. changes the `[link to Google](#)` to `[Go to top of page](#)`.<br>This line can now be an understandable test case for a link to a top-of-the-page anchor.

2. add another test case to the next line `[Go to Headers](#headers)` which tests a non-blank anchor and the markdown processing of a header.

3. tidies up some miscellaneous spelling and punctuation

![Test page changes](https://user-images.githubusercontent.com/66998419/139592198-e77a9552-d88c-4226-a8cc-ab7b130a7245.jpg)



## Verification

On a cloned copy of 
`https://github.com/corona-warn-app/cwa-website`

1. Execute:

```bash
npm ci
npm run test:prepare
npm start
```

2. Wait for web browser to start, then open http://localhost:8000/en/blog/2099-01-01-visual-test-page/
3. Scroll down to "Text & Bullet Points" and click on [Go to top of page](#).
4. Check that the browser goes to the top of the page.
5. Scroll down again to "Text & Bullet Points" and click on [Go to Headers](#headers).
6. Check that the browser goes to the **Headers** section
7. Open http://localhost:8000/de/blog/2099-01-01-visual-test-page/
8. Repeat above steps 3-6
9. Close the browser and kill the session / terminal running `npm start`

10. Execute:
```bash
npm run test:open
```
11. After Cypress starts, click on test `check_anchor_links.js` in Cypress console to run it

12. Confirm that `check_anchor_links.js` passes all tests
13. Click "Stop" in Cypress console
14. Click "Run 6 integration tests" in Cypress console
15. Wait for test completion and check that all tests have passed

~~Note that currently issue https://github.com/corona-warn-app/cwa-website/issues/1986 prevents Cypress test `check_anchor_links.js` completing successfully.~~
